presto-main/src/main/java/io/prestosql/sql/planner/iterative/rule/PushProjectionIntoTableScan.java

```diff
@@ -102,6 +103,9 @@ public Result apply(ProjectNode project, Captures captures, Context context)
             return Result.empty();
         }
 
+        List<ConnectorExpression> newConnectorProjections = result.get().getProjections();
+        checkState(newConnectorProjections.size() == projections.size(), "mismatch between input and output projections from the connector");
```

Comment:
Capitalize "Mismatch", include list sizes in the error message.

```suggestion
        checkState(newConnectorProjections.size() == projections.size(), "Mismatch between input and output projections from the connector: expected %s but got %s", projections.size(), newConnectorProjections.size());
```

(maybe you need to rewrap this)